src/python/pants/option/options_fingerprinter.py

```diff
@@ -81,9 +76,7 @@ def fingerprint(self, option_type, option_val):
         if not isinstance(option_val, (list, tuple, dict)):
             option_val = [option_val]
 
-        if option_type == target_option:
-            return self._fingerprint_target_specs(option_val)
```

Comment:
We were never hitting this code path in v2. We only ever had `self._build_graph` defined in v1 code, and not in the Pantsd code. So, I think this is safe to delete.